source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -358,6 +363,26 @@ void DnsCacheImpl::finishResolve(const std::string& host,
                                  absl::optional<MonotonicTime> resolution_time,
                                  bool is_proxy_lookup) {
   ASSERT(main_thread_dispatcher_.isThreadSafe());
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.dns_cache_set_ip_version_to_remove")) {
+    {
+      absl::MutexLock lock{&ip_version_to_remove_lock_};
+      if (ip_version_to_remove_.has_value()) {
+        if (config_.preresolve_hostnames_size() > 0) {
+          IS_ENVOY_BUG(
+              "Unable to delete IP version addresses when DNS preresolve hostnames are not empty.");
+          return;
```

Comment:
so in general if we hit an envoy bug we want to do best effort continue to function.  I'd think here we'd just skip the remove_if rather than possibly wedge state by not finishing up the resolutoin?

Reply:
Good idea. Done.